src/python/pants/backend/python/goals/setup_py.py

```diff
@@ -362,6 +372,50 @@ def validate_args(args: Tuple[str, ...]):
         raise InvalidSetupPyArgs("Cannot use the `upload` or `register` setup.py commands")
 
 
+@rule
+async def package_python_dist(
+    field_set: PythonDistributionFieldSet,
+    python_setup: PythonSetup,
+) -> BuiltPackage:
+
+    transitive_targets = await Get(TransitiveTargets, Addresses([field_set.address]))
+    exported_target = ExportedTarget(transitive_targets.roots[0])
+    interpreter_constraints = PexInterpreterConstraints.create_from_compatibility_fields(
+        (
+            tgt[PythonInterpreterCompatibility]
+            for tgt in transitive_targets.dependencies
+            if tgt.has_field(PythonInterpreterCompatibility)
+        ),
+        python_setup,
+    )
+    chroot = await Get(
+        SetupPyChroot,
+        SetupPyChrootRequest(exported_target, py2=interpreter_constraints.includes_python2()),
+    )
+
+    # If commands were provided, run setup.py with them; Otherwise just dump chroots.
+    if exported_target.target.has_field(SetupPyCommandsField):
+        setup_py_result = await Get(
+            RunSetupPyResult,
+            RunSetupPyRequest(
+                exported_target, chroot, exported_target.target[SetupPyCommandsField].value or ()
+            ),
+        )
+        dist_snapshot = await Get(Snapshot, Digest, setup_py_result.output)
+        return BuiltPackage(
+            setup_py_result.output,
+            relpaths=dist_snapshot.files,
+        )
+
+    else:
+        dirname = f"{chroot.setup_kwargs.name}-{chroot.setup_kwargs.version}"
```

Comment:
How would a user get this behavior through their BUILD file? Leave off the `setup_py_commands` field? If so, that'd be good to add to the documentation for the field.

Reply:
Yes and done.